code/__DEFINES/traits.dm

```diff
@@ -118,6 +118,11 @@
 		: FALSE)
 #define HAS_TRAIT_NOT_FROM(target, trait, source) (target.status_traits ? (target.status_traits[trait] ? (length(target.status_traits[trait] - source) > 0) : FALSE) : FALSE)
 
+/// Adds a trait, with the source being the caller's type.
+#define ADD_TYPED_TRAIT(target, trait)	ADD_TRAIT(target, trait, "[type]")
+/// Removes all traits added by ADD_TYPED_TRAIT.
+#define REMOVE_TYPED_TRAITS(target)		REMOVE_TRAITS_IN(target, "[type]")
```

Comment:
What do you mean? The source will just... be the type of whatever _calls_ it?

Reply:
Yea, that is confusing for such a generic since it is coming from a place outside of the define. Ideally we should also pass in the object that we explicitly want use as the type rather than using src. I can see mistakes being made here if someone tries to use this in the future.